syntax = "proto3";

package bloock;

import "bloock.proto";

service Config {
    rpc SetApiHost (SetApiHostRequest) returns (ConfigResponse);
    rpc SetNetworkConfig (SetNetworkConfigRequest) returns (ConfigResponse);
}

message ConfigResponse {
    optional Error error = 1;
}

message SetApiHostRequest {
    string host = 1;
}

message SetNetworkConfigRequest {
    Network network = 1;
    NetworkConfig config = 2;
}

enum Network {
    ETHEREUM_MAINNET = 0;
    ETHEREUM_RINKEBY = 1;
    BLOOCK_CHAIN = 2;
}

message NetworkConfig {
    string ContractAddress = 1;
    string ContractAbi = 2;
    string HttpProvider = 3;
}
